### PR TITLE
VP-6698: Fix inventory product amount after order has been created

### PR DESCRIPTION
**Pull request by AliveMen**
### Related task:     
VP-6698 Product's "In stock" amount not changing after order
### Problem:
When setting "Adjust inventory for orders" is enabled and user create order then "in stock" amount of that product not changing
### Changes:
Run CompareTo action synchronously

### CODE REVIEW

**Comment by akak1977** [comment]
Could you explain, why we should do this synchronously?

File: src/VirtoCommerce.OrdersModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs [hunk 1]
```diff
@@ -123,15 +123,15 @@ public virtual async Task<ProductInventoryChange[]> GetProductInventoryChangesFo
                     var itemChange = AbstractTypeFactory<ProductInventoryChange>.TryCreateInstance();
                     itemChange.ProductId = changedItem.ProductId;
                     itemChange.QuantityDelta = newQuantity - oldQuantity;
-                    itemChange.FulfillmentCenterId = await GetFullfilmentCenterForLineItemAsync(changedItem, customerOrder.StoreId, customerOrderShipments);
```
Comment (kostyrin) [inline src/VirtoCommerce.OrdersModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs:126]: look at inside the method GetFullfilmentCenterForLineItemAsync. Could you get Store and InventoryInfo asynchronously  outside the method? I think the could be refactor and change CompareTo to syncronously.

Comment (kostyrin) [reply src/VirtoCommerce.OrdersModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs:126]: also each time to get InventoryInfo by `GetProductsInventoryInfosAsync` could expensive operation.

File: src/VirtoCommerce.OrdersModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs [hunk 2]
```diff
@@ -95,7 +95,7 @@ public async Task TryAdjustOrderInventoryBackgroundJob(ProductInventoryChange[]
         /// </summary>
         /// <param name="changedEntry">The entry that describes changes made to order.</param>
         /// <returns>Array of required product inventory changes.</returns>
-        public virtual async Task<ProductInventoryChange[]> GetProductInventoryChangesFor(GenericChangedEntry<CustomerOrder> changedEntry)
```
[the remark this answers is not in the document]
Comment (AliveMen) [reply src/VirtoCommerce.OrdersModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs]: reverted

File: src/VirtoCommerce.OrdersModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs (hunk 1 above)
Comment (AliveMen) [reply src/VirtoCommerce.OrdersModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs:126]: Method refactored

File: src/VirtoCommerce.OrdersModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs [hunk 3]
```diff
@@ -244,10 +254,11 @@ protected virtual async Task AdjustInventory(IEnumerable<InventoryInfo> inventor
         /// Returns a fulfillment center identifier much suitable for given order line item
         /// </summary>
         /// <param name="lineItem"></param>
-        /// <param name="orderStoreId"></param>
         /// <param name="orderShipments"></param>
+        /// <param name="inventoryInfos"></param>
+        /// <param name="store"></param>
         /// <returns></returns>
-        protected virtual async Task<string> GetFullfilmentCenterForLineItemAsync(LineItem lineItem, string orderStoreId, Shipment[] orderShipments)
+        protected virtual string GetFullfilmentCenterForLineItemAsync(LineItem lineItem, Shipment[] orderShipments, List<InventoryInfo> inventoryInfos, Store store)
```
Comment (kostyrin) [inline src/VirtoCommerce.OrdersModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs:261]: rename to GetFullfilmentCenterForLineItem

Comment (AliveMen) [reply src/VirtoCommerce.OrdersModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs]: done

**Review by kostyrin — APPROVED** [review]

**Review by akak1977 — APPROVED** [review]

File: src/VirtoCommerce.OrdersModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs [hunk 4]
```diff
@@ -123,15 +127,15 @@ public virtual async Task<ProductInventoryChange[]> GetProductInventoryChangesFo
                     var itemChange = AbstractTypeFactory<ProductInventoryChange>.TryCreateInstance();
                     itemChange.ProductId = changedItem.ProductId;
                     itemChange.QuantityDelta = newQuantity - oldQuantity;
-                    itemChange.FulfillmentCenterId = await GetFullfilmentCenterForLineItemAsync(changedItem, customerOrder.StoreId, customerOrderShipments);
+                    itemChange.FulfillmentCenterId = GetFullfilmentCenterForLineItem(changedItem, customerOrderShipments, inventoryInfos, store);
```
Comment (tatarincev) [inline src/VirtoCommerce.OrdersModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs:130]: Please move the call `GetFullfilmentCenterForLineItem` out of this lambda function body  to line 134.

Comment (tatarincev) [reply src/VirtoCommerce.OrdersModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs:130]: that's will be enough to solve this issue

**Comment by AliveMen** [comment]
Another solution in https://github.com/VirtoCommerce/vc-module-order/pull/216

**Pull request closed without merging**